### PR TITLE
Pagejs refactoring

### PR DESCRIPTION
**Pull request by SimonLab**
see #189 
you can see the start of the documentation on [page.js](https://github.com/TheScienceMuseum/collectionsonline/wiki/Page.js) and [search](https://github.com/TheScienceMuseum/collectionsonline/wiki/Search)

### CODE REVIEW

File: client/lib/convert-url.js [hunk 1]
```diff
@@ -1,23 +0,0 @@
-var QueryString = require('querystring');
```
Comment (SimonLab) [inline client/lib/convert-url.js]: This file was use to convert url between a json and url format for the ajax request. We only use a html to json format url which is simplified with the file to-json-url.js

**Comment by SimonLab** [comment]
@nelsonic @Danwhy comments are welcome, thanks

File: client/middleware/error-404.js [hunk 2]
```diff
@@ -1,12 +1,24 @@
+var Templates = require('../templates');
+
 module.exports = function (page) {
-  page(function (ctx, next) {
+  // page(function (ctx, next) {
+  //   next();
+  // });
+  page(render);
+  function render (ctx, next) {
     console.warn('Page not found', ctx.path);
```
Comment (nelsonic) [inline client/middleware/error-404.js]: Do we _always_ `console.warn` ?

File: client/routes/home.js [hunk 3]
```diff
@@ -2,35 +2,30 @@ var svg4everybody = require('svg4everybody');
 var $ = require('jquery');
 var QueryString = require('querystring');
 var searchBox = require('../lib/search-box');
-var createQueryParams = require('../../lib/query-params.js');
-var getData = require('../lib/get-data.js');
+var data = require('../../fixtures/data');
```
Comment (nelsonic) [inline client/routes/home.js]: is this _temporarily_ rendering _dummy data_?

File: client/middleware/error-404.js (hunk 2 above)
Comment (SimonLab) [inline client/middleware/error-404.js]: I think we use console.war only to display which file couldn't be loaded and when a document of elasticsaerch doesn't belong to any of the main type (people, documents or object)

File: client/routes/home.js (hunk 3 above)
Comment (SimonLab) [inline client/routes/home.js]: yes currently there are no dynamic data on the home page

Comment (nelsonic) [inline client/routes/home.js]: That's fine (_for now_) because it will load faster. 🐎

**Comment by nelsonic** [comment]
Looks good. 👍

**Comment by Danwhy** [comment]
LGTM

**Pull request merged**